cdap-proto/src/main/java/co/cask/cdap/proto/id/EntityId.java

```diff
@@ -82,6 +82,25 @@ public final EntityType getEntity() {
     return (T) id.toId();
   }
 
+  public static <T extends EntityId> T fromString(String string) {
```

Comment:
No autocompletion right now. We need to add autocompletion for list-style arguments and EntityId.